File: ClassRelation/Dependency.py
class Car:
    @staticmethod
    def run():
        print("Car is running")


class Driver:
    # 通過方法的參數
    def drive1(self, car):
        car.run()

    # 通過局部變數
    def drive2(self):
        Car().run()

    # 對静態方法的調用
    @staticmethod
    def drive3():
        Car.run()


if __name__ == "__main__":
    car = Car()
    Driver().drive1(car)
    Driver().drive2()
    Driver().drive3()

File: ClassRelation/Dependency.cpp
#include<iostream>
class Car{
    public:
        static void run(){
            std::cout<<"Car is running"<<std::endl;
        }
};
class Driver{
public:
    // B作為A的成員函數參數
    void drive1(Car car){
        car.run();
    }
    // B作為A的成員函數的局部變量
    void drive2(){
        Car car ;
        car.run();
    }
    // A的成員函數調用B的靜態方法
    void drive3(){
        Car::run();
    }
};
int main(){
    Car car;
    Driver().drive1(car);
    Driver().drive2();
    Driver().drive3();
    return 0;
}